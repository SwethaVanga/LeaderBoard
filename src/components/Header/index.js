import React from 'react'
import style from './index.module.scss'
import People from '../../assets/images/business-people.svg'
import { ReactSVG } from 'react-svg'
import User from '../../shared/components/User'

const Header = ({ users }) => {
  return (
    <div className={style.header}>
      <div className={style.header__rating}>
        <span className={style.header_title}>All time Highest Scorers</span>
        <ul className={style.header__list}>
          {!!users.length &&
            users.map((user, index) => {
              return <User name={user.name} score={user.score} key={index} />
            })}
        </ul>
      </div>
      <div className={style.header__image}>
        <ReactSVG src={People} className={style.image} />
      </div>
    </div>
  )
}

export default Header
