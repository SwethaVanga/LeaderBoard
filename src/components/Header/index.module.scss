@import "../../assets/style/variables.scss";

.header {
  display: flex;
  justify-content: space-between;
  width: 100%;
  height: 187px;
  padding: 16px 27px;
  margin-bottom: 27px;
  border-radius: 10px;
  border: 1px solid $header-border;
  background-color: $header;
  box-sizing: border-box;
  overflow: hidden;
  position: relative;

  &_title {
    font-family: "Gilroy";
    font-style: normal;
    font-weight: 600;
    font-size: 20px;
    line-height: 32px;
    color: $white;
  }

  &__list {
    display: flex;
    justify-content: flex-start;
    width: 498px;
    padding: 0;
  }

  &__image {
    position: absolute;
    right: 0;
    top: 7px;
  }
}

.image {
  & svg {
    width: 390px;
    height: 248px;

    & path:nth-child(1) {
      fill: transparent;
    }
    & circle:nth-child(2) {
      fill: transparent;
    }
    & path:nth-child(3) {
      fill: transparent;
    }
    & path:nth-child(4) {
      fill: transparent;
    }
    & path:nth-child(5) {
      fill: $header-border;
    }
    & path:nth-child(6) {
      fill: $line;
    }
    & path:nth-child(7) {
      fill: $line;
    }
    & path:nth-child(9) {
      fill: transparent;
    }
    & path:nth-child(8) {
      fill: $oval;
    }
    & path:nth-child(10) {
      fill: transparent;
    }
    & path:nth-child(11) {
      fill: transparent;
    }
    & path:nth-child(12) {
      fill: transparent;
    }
    & path:nth-child(13) {
      fill: $header-border;
    }
    & circle:nth-child(14) {
      fill: $header-border;
    }
    & circle:nth-child(15) {
      fill: $header-border;
    }
    & path:nth-child(16) {
      fill: $header-border;
    }
    & path:nth-child(17) {
      fill: $header-border;
    }
    & path:nth-child(18) {
      fill: $line;
    }
    & path:nth-child(19) {
      fill: $line;
    }
    & path:nth-child(20) {
      fill: $line;
    }
    & path:nth-child(21) {
      fill: $line;
    }
    & path:nth-child(22) {
      fill: $header-border;
    }
    & path:nth-child(23) {
      fill: $oval;
    }
    & path:nth-child(24) {
      fill: $header-border;
    }
    & path:nth-child(33) {
      fill: $oval;
    }
    & path:nth-child(35) {
      fill: $oval;
    }
    & circle:nth-child(17) {
      fill: $oval;
    }
  }
}
