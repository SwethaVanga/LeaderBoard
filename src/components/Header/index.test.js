import React from 'react'
import renderer from 'react-test-renderer'
import Header from './index.js'

it('render correctly header component', () => {
  const props = [{ name: '', score: 0 }]
  const HeaderComponent = renderer.create(<Header users={props} />).toJSON()
  expect(HeaderComponent).toMatchSnapshot()
})
