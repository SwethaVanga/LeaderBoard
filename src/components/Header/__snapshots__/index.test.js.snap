// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly header component 1`] = `
<div
  className="header"
>
  <div
    className="header__rating"
  >
    <span
      className="header_title"
    >
      All time Highest Scorers
    </span>
    <ul
      className="header__list"
    >
      <div
        className="user"
      >
        <div
          className="user__avatar"
        >
          <div
            className="user__image"
          >
            <img
              alt=""
              src="chris.svg"
            />
          </div>
          <div
            className="user__score"
          >
            0
          </div>
        </div>
        <div
          className="user__name"
        >
          
        </div>
      </div>
    </ul>
  </div>
  <div
    className="header__image"
  >
    <div
      className="image"
    />
  </div>
</div>
`;
